docs/guides/index.rst

```diff
@@ -14,6 +14,7 @@ Guides
    extracting_intermediates
    lr_schedule
    use_checkpointing
+   spmd_api
```

Comment:
Maybe this is an advanced guide? It looks quite involved and you do a number of advanced things (lifted transformations, scan over layers, ...)

Reply:
Lifted transformations is the only advanced topic here, and it is not really explored in detail. Since JAX team may want to upstream `pjit` to `jit` in the future, it may make sense to put it on the later section of guides as an introduction on how to scale up a model. (JAX also put their [pjit doc](https://jax.readthedocs.io/en/latest/jax-101/08-pjit.html) as the last piece of their 101 tutorial.)